modules/portmaster/projects/safing/portmaster-api/src/lib/netquery.service.ts

```diff
@@ -213,8 +213,10 @@ export class Netquery {
     @Inject(PORTMASTER_HTTP_API_ENDPOINT) private httpAPI: string,
   ) { }
 
-  query(query: Query): Observable<QueryResult[]> {
-    return this.http.post<{ results: QueryResult[] }>(`${this.httpAPI}/v1/netquery/query`, query)
+  query(query: Query, origin: string): Observable<QueryResult[]> {
+    return this.http.post<{ results: QueryResult[] }>(`${this.httpAPI}/v1/netquery/query`, query, {
+      params: new HttpParams().set("origin", origin)
```

Comment:
Please use a custom parameter, I fear that using "Origin" might come back to haunt us later.